tests/unit/utils/recursive-keys-test.js

```diff
@@ -0,0 +1,152 @@
+import recursiveKeys from 'ember-changeset/utils/recursive-keys';
+import { module, test } from 'qunit';
+
+import EmberObject from '@ember/object';
+
+module('Unit | Utility | recursive-keys');
+
+test('it returns an array with keys as nested strings', function(assert) {
```

Comment:
Since we're repeating the same assertion, let's write these tests in table-driven form. Example:

```js
let testData = [
  {
    desc: 'it returns an array with keys as nested strings',
    input: {
      topValue: -1,
      item: {
        secondValue: 0,
        innerItem: {
        thing: 1,
        other: 2,
        nest: {
          thing: 3,
          other: 4
        }
      }
    },
    expectedResult: [
      'topValue',
      'item.secondValue',
      'item.innerItem.thing',
      'item.innerItem.other',
      'item.innerItem.nest.thing',
      'item.innerItem.nest.other',
    ]
  },

  // ...
];

testData.forEach(({ desc, input, expected }) => {
  test(desc, function(assert) {
    assert.deepEqual(leafKeys(input), expected, desc);
  });
});
```